File: frontend/src/types/recipe.ts
import { Timestamp } from 'firebase/firestore';
import { MealType } from './meal';

export interface Recipe {
    id: string;
    name: string;
    instructions: string;
    createdAt: Timestamp;
    photos: string[];
    nutritionalValues: NutritionalValues;
    parentRecipeId: string | null;
}

export interface NutritionalValues {
    calories: number;
    protein: number;
    fat: number;
    carbs: number;
}

export interface RecipeReference {
    recipeId: string;
    dietId: string;
    userId: string;
    mealType: MealType;
    addedAt: Timestamp;
}

File: app/src/main/java/com/noisevisionsoftware/szytadieta/domain/model/health/dietPlan/Recipe.kt
package com.noisevisionsoftware.szytadieta.domain.model.health.dietPlan

import com.google.firebase.Timestamp
import com.google.firebase.firestore.PropertyName

data class Recipe(
    val id: String = "",
    val name: String = "",
    val instructions: String = "",
    val createdAt: Timestamp?,
    val photos: List<String> = emptyList(),
    val nutritionalValues: NutritionalValues? = null,
    val ingredients: List<String> = emptyList(),
    val parentRecipeId: String? = null
)